packages/backend/src/managers/llama-stack/llamaStackManager.spec.ts

```diff
@@ -0,0 +1,181 @@
+});
+
+beforeEach(() => {
+  const containerRegistry = new ContainerRegistry();
+  containerRegistry.init();
+  llamaStackManager = new LlamaStackManager(
+    '',
+    taskRegistry,
+    podmanConnection,
+    containerRegistry,
+    configurationRegistry,
+    telemetryMock,
+  );
+  llamaStackManager.init();
+  taskRegistry.deleteByLabels({ trackingId: LLAMA_STACK_CONTAINER_TRACKINGID });
+});
+
+test('getLlamaStackContainer should return undefined if no containers', async () => {
+  vi.mocked(containerEngine.listContainers).mockResolvedValue([]);
+  const containerId = await llamaStackManager.getLlamaStackContainer();
+  expect(containerId).toBeUndefined();
+});
+
+test('getLlamaStackContainer should return undefined if no llama stack container', async () => {
+  vi.mocked(containerEngine.listContainers).mockResolvedValue([{ Id: 'dummyId' } as unknown as ContainerInfo]);
+  const containerId = await llamaStackManager.getLlamaStackContainer();
+  expect(containerId).toBeUndefined();
+});
+
+test('getLlamaStackContainer should return id if instructlab container', async () => {
+  vi.mocked(containerEngine.listContainers).mockResolvedValue([
+    {
+      Id: 'dummyId',
+      State: 'running',
+      Labels: {
+        [LLAMA_STACK_CONTAINER_LABEL]: 'dummyLabel',
+        [LLAMA_STACK_API_PORT_LABEL]: '50000',
+      },
+    } as unknown as ContainerInfo,
+  ]);
+  const containerInfo = await llamaStackManager.getLlamaStackContainer();
+  expect(containerInfo).toEqual({ containerId: 'dummyId', port: 50000 });
+});
+
+test('requestCreateLlamaStackContainer throws error if no podman connection', async () => {
+  const containerIdPromise = llamaStackManager.requestCreateLlamaStackContainer({});
+  await expect(containerIdPromise).rejects.toBeInstanceOf(Error);
+});
+
+async function waitTasks(id: string, nb: number): Promise<Task[]> {
+  return vi.waitFor(() => {
+    const tasks = taskRegistry.getTasksByLabels({ trackingId: id });
+    if (tasks.length !== nb) {
+      throw new Error('not completed');
+    }
+    return tasks;
+  });
+}
+
+test('requestCreateLlamaStackContainer returns id and error if listImage returns error', async () => {
+  vi.mocked(podmanConnection.findRunningContainerProviderConnection).mockReturnValue({
+    name: 'Podman Machine',
+    vmType: VMType.UNKNOWN,
+    type: 'podman',
+    status: () => 'started',
+    endpoint: {
+      socketPath: 'socket.sock',
+    },
+  });
+  vi.mocked(containerEngine.listImages).mockRejectedValue(new Error());
+  await llamaStackManager.requestCreateLlamaStackContainer({});
+  const tasks = await waitTasks(LLAMA_STACK_CONTAINER_TRACKINGID, 2);
+  expect(tasks.some(task => task.state === 'error')).toBeTruthy();
+});
+
+test('requestCreateLlamaStackContainer returns id and error if listImage returns image', async () => {
+  vi.mocked(podmanConnection.findRunningContainerProviderConnection).mockReturnValue({
+    name: 'Podman Machine',
```

Comment:
to avoid repetition in all tests, please declare a constant in the top of the file

https://github.com/containers/podman-desktop-extension-ai-lab/blob/e34d59f8e3ab8ed84fb154827dbcf46313d07de5/packages/backend/src/workers/provider/LlamaCppPython.spec.ts#L69

(if you could use only uppercase for it,  that would be even better 🙏 )